tests/test_auto_offer_on_device_rest.py

```diff
@@ -22,16 +22,16 @@ def on_market_cycle(self, market_info):
         :param market_info: Incoming message containing the newly-created market info
         :return: None
         """
-        if self.registered is False:
+        if self.registered is False or self.is_finished is True:
             return
         logging.debug(f"New market information {market_info}")
         if "available_energy_kWh" in market_info["device_info"] and market_info["device_info"]["available_energy_kWh"] > 0.0:
             offer = self.offer_energy_rate(market_info["device_info"]["available_energy_kWh"], 16)
             logging.debug(f"Offer placed on the new market: {offer}")
             assert len(self.list_offers()) == 1
         if "energy_requirement_kWh" in market_info["device_info"] and market_info["device_info"]["energy_requirement_kWh"] > 0.0:
-            bid = self.bid_energy(market_info["device_info"]["energy_requirement_kWh"], 30)
-            logging.error(f"Bid placed on the new market: {bid}")
+            bid = self.bid_energy(market_info["device_info"]["energy_requirement_kWh"] / 2, 30)
```

Comment:
Can you explain the reason to place bid with half of required energy?

Reply:
I guess it is just a simple example to make it more visual on the graphs, in order to check the values in the ui afterwards

Reply:
Okay, sounds reasonable